cartographer_grpc/local_trajectory_uploader.cc

```diff
@@ -69,57 +69,88 @@ void LocalTrajectoryUploader::ProcessSendQueue() {
   while (!shutting_down_) {
     auto data_message = send_queue_.PopWithTimeout(kPopTimeout);
     if (data_message) {
-      if (const auto *fixed_frame_pose_data =
-              dynamic_cast<const proto::AddFixedFramePoseDataRequest *>(
+      if (auto *fixed_frame_pose_data =
+              dynamic_cast<proto::AddFixedFramePoseDataRequest *>(
                   data_message.get())) {
         ProcessFixedFramePoseDataMessage(fixed_frame_pose_data);
-      } else if (const auto *imu_data =
-                     dynamic_cast<const proto::AddImuDataRequest *>(
-                         data_message.get())) {
+      } else if (auto *imu_data = dynamic_cast<proto::AddImuDataRequest *>(
+                     data_message.get())) {
         ProcessImuDataMessage(imu_data);
-      } else if (const auto *odometry_data =
-                     dynamic_cast<const proto::AddOdometryDataRequest *>(
+      } else if (auto *odometry_data =
+                     dynamic_cast<proto::AddOdometryDataRequest *>(
                          data_message.get())) {
         ProcessOdometryDataMessage(odometry_data);
+      } else if (auto *local_slam_result_data =
+                     dynamic_cast<proto::AddLocalSlamResultDataRequest *>(
+                         data_message.get())) {
+        ProcessLocalSlamResultDataMessage(local_slam_result_data);
       } else {
         LOG(FATAL) << "Unknown message type: " << data_message->GetTypeName();
       }
     }
   }
 }
 
+void LocalTrajectoryUploader::ProcessSensorMetadata(
```

Comment:
Can you come up with a more descriptive function name? Something that conveys that you are translating from local trajectory IDs to cloud ones?

Reply:
Done